server/eclipse-project/src/test/java/us/freeandfair/corla/controllers/BallotSelectionTest.java

```diff
@@ -48,18 +39,46 @@ public class BallotSelectionTest {
 
   private Boolean return_cvr = true;
 
-  @BeforeTest()
-  public void setUp() {
-    Setup.setProperties();
+  /**
+   * Container for the mock-up database.
+   */
+  static PostgreSQLContainer<?> postgres
+          = new PostgreSQLContainer<>("postgres:15-alpine")
+          // None of these actually have to be the same as the real database (except its name), but this
+          // makes it easy to match the setup scripts.
+          .withDatabaseName("corla")
+          .withUsername("corlaadmin")
+          .withPassword("corlasecret")
+          // .withInitScripts("corlaInit.sql","contest.sql");
+          .withInitScript("SQL/corlaInitEmpty.sql");
+
+  @BeforeClass
+  public static void beforeAll() {
+    postgres.start();
+    Properties hibernateProperties = new Properties();
+    hibernateProperties.setProperty("hibernate.driver", "org.postgresql.Driver");
+    hibernateProperties.setProperty("hibernate.url", postgres.getJdbcUrl());
+    hibernateProperties.setProperty("hibernate.user", postgres.getUsername());
+    hibernateProperties.setProperty("hibernate.pass", postgres.getPassword());
+    hibernateProperties.setProperty("hibernate.dialect", "org.hibernate.dialect.PostgreSQL9Dialect");
+    Persistence.setProperties(hibernateProperties);
     Persistence.beginTransaction();
+
```

Comment:
@michelleblom pointed out when I set all this up explicitly that there's probably a much slicker way to do it - possibly setting all these values in test/resources/test.properties.

This worked great in the raire-service: https://github.com/DemocracyDevelopers/raire-service/tree/main/src/test/resources
but we were not keen to fiddle with colorado-rla in case we messed up the other tests. But now that you've got the other tests working, possibly you/we can avoid doing all this postgresql container config explicitly every time.

Reply:
Ah thanks, that's a great idea.